platform/windows/windows_terminal_logger.cpp

```diff
@@ -81,76 +81,51 @@ void WindowsTerminalLogger::log_error(const char *p_function, const char *p_file
 	if (!hCon || hCon == INVALID_HANDLE_VALUE) {
 		StdLogger::log_error(p_function, p_file, p_line, p_code, p_rationale, p_type);
 	} else {
-		CONSOLE_SCREEN_BUFFER_INFO sbi; //original
-		GetConsoleScreenBufferInfo(hCon, &sbi);
-
-		WORD current_bg = sbi.wAttributes & (BACKGROUND_RED | BACKGROUND_GREEN | BACKGROUND_BLUE | BACKGROUND_INTENSITY);
-
-		uint32_t basecol = 0;
-		switch (p_type) {
-			case ERR_ERROR:
-				basecol = FOREGROUND_RED;
-				break;
-			case ERR_WARNING:
-				basecol = FOREGROUND_RED | FOREGROUND_GREEN;
-				break;
-			case ERR_SCRIPT:
-				basecol = FOREGROUND_RED | FOREGROUND_BLUE;
-				break;
-			case ERR_SHADER:
-				basecol = FOREGROUND_GREEN | FOREGROUND_BLUE;
-				break;
+		const char *err_details;
+		if (p_rationale && p_rationale[0]) {
+			err_details = p_rationale;
+		} else {
+			err_details = p_code;
 		}
 
-		basecol |= current_bg;
+		// Disable color codes if stdout is not a TTY.
+		// This prevents Godot from writing ANSI escape codes when redirecting
+		// stdout and stderr to a file.
+		//
+		// FIXME: Is there a way to check whether stdout is a TTY on Windows?
+		const char *red = "\E[0;91m";
```

Comment:
We should probably avoid using `\E` as our escape code, as it doesn't actually appear in the standard[^1]. All existing instances can be converted to `\x1b` or `\033` instead, similar to what we use in SCons
```suggestion
		const char *red = "\x1b[0;91m";
```
[^1]: https://en.wikipedia.org/wiki/Escape_sequences_in_C#endnote_Note1